{
    "Modified": "2018-09-24T20:29:00.703000",
    "Published": "2018-09-24T20:29:00.703000",
    "capec": [],
    "cvss": 5.0,
    "cwe": "CWE-399",
    "id": "CVE-2018-14647",
    "last-modified": "2018-12-20T09:32:55.670000",
    "nessus": [
        {
            "NASL family": "Debian Local Security Checks",
            "NASL id": "DEBIAN_DSA-4306.NASL",
            "description": "Multiple security issues were discovered in Python: ElementTree failed\nto initialise Expat's hash salt, two denial of service issues were\nfound in difflib and poplib and the shutil module was affected by a\ncommand injection vulnerability.",
            "last seen": "2019-01-16",
            "modified": "2018-12-19",
            "plugin id": "117812",
            "published": "2018-09-28",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=117812",
            "title": "Debian DSA-4306-1 : python2.7 - security update"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2018-4544E8DBC8.NASL",
            "description": "Security fix for CVE-2018-14647\n\nNote that Tenable Network Security has extracted the preceding\ndescription block directly from the Fedora update system website.\nTenable has attempted to automatically clean and format it as much as\npossible without introducing additional issues.",
            "last seen": "2019-01-16",
            "modified": "2019-01-03",
            "plugin id": "120386",
            "published": "2019-01-03",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=120386",
            "title": "Fedora 29 : python34 (2018-4544e8dbc8)"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2018-9860917DB0.NASL",
            "description": "Update to 3.7.1, Security fix for CVE-2018-14647\n\nNote that Tenable Network Security has extracted the preceding\ndescription block directly from the Fedora update system website.\nTenable has attempted to automatically clean and format it as much as\npossible without introducing additional issues.",
            "last seen": "2019-01-16",
            "modified": "2019-01-03",
            "plugin id": "120640",
            "published": "2019-01-03",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=120640",
            "title": "Fedora 29 : python3 (2018-9860917db0)"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2018-D3B53D81E6.NASL",
            "description": "Security fix for CVE-2018-14647\n\nNote that Tenable Network Security has extracted the preceding\ndescription block directly from the Fedora update system website.\nTenable has attempted to automatically clean and format it as much as\npossible without introducing additional issues.",
            "last seen": "2019-01-16",
            "modified": "2019-01-03",
            "plugin id": "120821",
            "published": "2019-01-03",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=120821",
            "title": "Fedora 28 : python26 (2018-d3b53d81e6)"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2018-EE97FC9E81.NASL",
            "description": "Security fix for CVE-2018-14647\n\nNote that Tenable Network Security has extracted the preceding\ndescription block directly from the Fedora update system website.\nTenable has attempted to automatically clean and format it as much as\npossible without introducing additional issues.",
            "last seen": "2019-01-16",
            "modified": "2019-01-03",
            "plugin id": "120887",
            "published": "2019-01-03",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=120887",
            "title": "Fedora 29 : python2 (2018-ee97fc9e81)"
        },
        {
            "NASL family": "SuSE Local Security Checks",
            "NASL id": "SUSE_SU-2018-3156-1.NASL",
            "description": "This update for python fixes the following issue :\n\nCVE-2018-14647: Python's elementtree C accelerator failed to\ninitialise Expat's hash salt during initialization. This could make it\neasy to conduct denial of service attacks against Expat by\nconstructing an XML document that would cause pathological hash\ncollisions in Expat's internal data structures, consuming large\namounts CPU and RAM (bsc#1109847)\n\nNote that Tenable Network Security has extracted the preceding\ndescription block directly from the SUSE security advisory. Tenable\nhas attempted to automatically clean and format it as much as possible\nwithout introducing additional issues.",
            "last seen": "2019-01-16",
            "modified": "2018-12-21",
            "plugin id": "118171",
            "published": "2018-10-17",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=118171",
            "title": "SUSE SLES11 Security Update : python (SUSE-SU-2018:3156-1)"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2018-A2C1453607.NASL",
            "description": "Security fix for CVE-2018-14647\n\nNote that Tenable Network Security has extracted the preceding\ndescription block directly from the Fedora update system website.\nTenable has attempted to automatically clean and format it as much as\npossible without introducing additional issues.",
            "last seen": "2019-01-16",
            "modified": "2018-12-21",
            "plugin id": "118410",
            "published": "2018-10-26",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=118410",
            "title": "Fedora 27 : python35 (2018-a2c1453607)"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2018-7689556AB2.NASL",
            "description": "Security fix for CVE-2018-14647 (#1631822)\n\nNote that Tenable Network Security has extracted the preceding\ndescription block directly from the Fedora update system website.\nTenable has attempted to automatically clean and format it as much as\npossible without introducing additional issues.",
            "last seen": "2019-01-16",
            "modified": "2019-01-03",
            "plugin id": "120539",
            "published": "2019-01-03",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=120539",
            "title": "Fedora 28 : python35 (2018-7689556ab2)"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2018-2BF852F063.NASL",
            "description": "Security fix for CVE-2018-14647\n\nNote that Tenable Network Security has extracted the preceding\ndescription block directly from the Fedora update system website.\nTenable has attempted to automatically clean and format it as much as\npossible without introducing additional issues.",
            "last seen": "2019-01-16",
            "modified": "2019-01-03",
            "plugin id": "120315",
            "published": "2019-01-03",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=120315",
            "title": "Fedora 28 : python2 (2018-2bf852f063)"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2018-14526CBEBE.NASL",
            "description": "Security fix for CVE-2018-14647\n\nNote that Tenable Network Security has extracted the preceding\ndescription block directly from the Fedora update system website.\nTenable has attempted to automatically clean and format it as much as\npossible without introducing additional issues.",
            "last seen": "2019-01-16",
            "modified": "2018-12-21",
            "plugin id": "118409",
            "published": "2018-10-26",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=118409",
            "title": "Fedora 27 : python26 (2018-14526cbebe)"
        },
        {
            "NASL family": "Amazon Linux Local Security Checks",
            "NASL id": "AL2_ALAS-2018-1132.NASL",
            "description": "Python's elementtree C accelerator failed to initialise Expat's hash\nsalt during initialization. This could make it easy to conduct denial\nof service attacks against Expat by contructing an XML document that\nwould cause pathological hash collisions in Expat's internal data\nstructures, consuming large amounts CPU and RAM.(CVE-2018-14647)",
            "last seen": "2019-01-16",
            "modified": "2018-12-21",
            "plugin id": "119786",
            "published": "2018-12-20",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=119786",
            "title": "Amazon Linux 2 : python3 (ALAS-2018-1132)"
        },
        {
            "NASL family": "Amazon Linux Local Security Checks",
            "NASL id": "ALA_ALAS-2018-1132.NASL",
            "description": "Python's elementtree C accelerator failed to initialise Expat's hash\nsalt during initialization. This could make it easy to conduct denial\nof service attacks against Expat by contructing an XML document that\nwould cause pathological hash collisions in Expat's internal data\nstructures, consuming large amounts CPU and RAM.(CVE-2018-14647)",
            "last seen": "2019-01-16",
            "modified": "2019-01-14",
            "plugin id": "119812",
            "published": "2018-12-21",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=119812",
            "title": "Amazon Linux AMI : python34 / python36 (ALAS-2018-1132)"
        },
        {
            "NASL family": "Amazon Linux Local Security Checks",
            "NASL id": "ALA_ALAS-2018-1101.NASL",
            "description": "Python's elementtree C accelerator failed to initialise Expat's hash\nsalt during initialization. This could make it easy to conduct denial\nof service attacks against Expat by contructing an XML document that\nwould cause pathological hash collisions in Expat's internal data\nstructures, consuming large amounts CPU and RAM.(CVE-2018-14647)",
            "last seen": "2019-01-16",
            "modified": "2018-12-21",
            "plugin id": "118805",
            "published": "2018-11-08",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=118805",
            "title": "Amazon Linux AMI : python35 (ALAS-2018-1101)"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2018-5ED8FB9EFA.NASL",
            "description": "Update to 3.6.7\n\nSecurity fix for CVE-2018-14647\n\nNote that Tenable Network Security has extracted the preceding\ndescription block directly from the Fedora update system website.\nTenable has attempted to automatically clean and format it as much as\npossible without introducing additional issues.",
            "last seen": "2019-01-16",
            "modified": "2019-01-03",
            "plugin id": "120458",
            "published": "2019-01-03",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=120458",
            "title": "Fedora 28 : python3 (2018-5ed8fb9efa)"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2018-BBBD8CC3A6.NASL",
            "description": "Security fix for CVE-2018-14647\n\nNote that Tenable Network Security has extracted the preceding\ndescription block directly from the Fedora update system website.\nTenable has attempted to automatically clean and format it as much as\npossible without introducing additional issues.",
            "last seen": "2019-01-16",
            "modified": "2019-01-03",
            "plugin id": "120744",
            "published": "2019-01-03",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=120744",
            "title": "Fedora 28 : python33 (2018-bbbd8cc3a6)"
        },
        {
            "NASL family": "Debian Local Security Checks",
            "NASL id": "DEBIAN_DSA-4307.NASL",
            "description": "Multiple security issues were discovered in Python: ElementTree failed\nto initialise Expat's hash salt, two denial of service issues were\nfound in difflib and poplib and a buffer overflow in\nPyString_DecodeEscape.",
            "last seen": "2019-01-16",
            "modified": "2018-11-13",
            "plugin id": "117838",
            "published": "2018-10-01",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=117838",
            "title": "Debian DSA-4307-1 : python3.5 - security update"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2018-71FD5DB181.NASL",
            "description": "Security fix for CVE-2018-14647\n\nNote that Tenable Network Security has extracted the preceding\ndescription block directly from the Fedora update system website.\nTenable has attempted to automatically clean and format it as much as\npossible without introducing additional issues.",
            "last seen": "2019-01-16",
            "modified": "2019-01-03",
            "plugin id": "120526",
            "published": "2019-01-03",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=120526",
            "title": "Fedora 29 : python26 (2018-71fd5db181)"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2018-937E8A39C4.NASL",
            "description": "Updated to 3.6.7, including security fix for CVE-2018-14647.\n\nNote that Tenable Network Security has extracted the preceding\ndescription block directly from the Fedora update system website.\nTenable has attempted to automatically clean and format it as much as\npossible without introducing additional issues.",
            "last seen": "2019-01-16",
            "modified": "2019-01-03",
            "plugin id": "120624",
            "published": "2019-01-03",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=120624",
            "title": "Fedora 29 : python36 (2018-937e8a39c4)"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2018-49D6E4BC3F.NASL",
            "description": "Update to 3.7.1\n\n----\n\nSecurity fix for CVE-2018-14647 (#1631822)\n\nNote that Tenable Network Security has extracted the preceding\ndescription block directly from the Fedora update system website.\nTenable has attempted to automatically clean and format it as much as\npossible without introducing additional issues.",
            "last seen": "2019-01-16",
            "modified": "2019-01-03",
            "plugin id": "120395",
            "published": "2019-01-03",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=120395",
            "title": "Fedora 28 : python37 (2018-49d6e4bc3f)"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2018-AC14DBF3FD.NASL",
            "description": "Security fix for CVE-2018-14647\n\nNote that Tenable Network Security has extracted the preceding\ndescription block directly from the Fedora update system website.\nTenable has attempted to automatically clean and format it as much as\npossible without introducing additional issues.",
            "last seen": "2019-01-16",
            "modified": "2019-01-03",
            "plugin id": "120699",
            "published": "2019-01-03",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=120699",
            "title": "Fedora 29 : python35 (2018-ac14dbf3fd)"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2018-B6DE5FC905.NASL",
            "description": "Security fix for CVE-2018-14647\n\nNote that Tenable Network Security has extracted the preceding\ndescription block directly from the Fedora update system website.\nTenable has attempted to automatically clean and format it as much as\npossible without introducing additional issues.",
            "last seen": "2019-01-16",
            "modified": "2018-12-21",
            "plugin id": "118241",
            "published": "2018-10-22",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=118241",
            "title": "Fedora 27 : python2 (2018-b6de5fc905)"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2018-28EA2290AD.NASL",
            "description": "Security fix for CVE-2018-14647\n\nNote that Tenable Network Security has extracted the preceding\ndescription block directly from the Fedora update system website.\nTenable has attempted to automatically clean and format it as much as\npossible without introducing additional issues.",
            "last seen": "2019-01-16",
            "modified": "2018-12-21",
            "plugin id": "118896",
            "published": "2018-11-13",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=118896",
            "title": "Fedora 27 : python33 (2018-28ea2290ad)"
        },
        {
            "NASL family": "Ubuntu Local Security Checks",
            "NASL id": "UBUNTU_USN-3817-1.NASL",
            "description": "It was discovered that Python incorrectly handled large amounts of\ndata. A remote attacker could use this issue to cause Python to crash,\nresulting in a denial of service, or possibly execute arbitrary code.\nThis issue only affected Ubuntu 14.04 LTS and Ubuntu 16.04 LTS.\n(CVE-2018-1000030)\n\nIt was discovered that Python incorrectly handled running external\ncommands in the shutil module. A remote attacker could use this issue\nto cause Python to crash, resulting in a denial of service, or\npossibly execute arbitrary code. (CVE-2018-1000802)\n\nIt was discovered that Python incorrectly used regular expressions\nvulnerable to catastrophic backtracking. A remote attacker could\npossibly use this issue to cause a denial of service. This issue only\naffected Ubuntu 14.04 LTS and Ubuntu 16.04 LTS. (CVE-2018-1060,\nCVE-2018-1061)\n\nIt was discovered that Python failed to initialize Expat's hash salt.\nA remote attacker could possibly use this issue to cause hash\ncollisions, leading to a denial of service. (CVE-2018-14647).\n\nNote that Tenable Network Security has extracted the preceding\ndescription block directly from the Ubuntu security advisory. Tenable\nhas attempted to automatically clean and format it as much as possible\nwithout introducing additional issues.",
            "last seen": "2019-01-16",
            "modified": "2018-12-19",
            "plugin id": "118954",
            "published": "2018-11-14",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=118954",
            "title": "Ubuntu 14.04 LTS / 16.04 LTS / 18.04 LTS : python2.7, python3.4, python3.5 vulnerabilities (USN-3817-1)"
        }
    ],
    "ranking": [
        [
            {
                "circl": 3
            }
        ]
    ],
    "references": [
        "http://www.securityfocus.com/bid/105396",
        "http://www.securitytracker.com/id/1041740",
        "https://bugs.python.org/issue34623",
        "https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-14647",
        "https://usn.ubuntu.com/3817-1/",
        "https://usn.ubuntu.com/3817-2/",
        "https://www.debian.org/security/2018/dsa-4306",
        "https://www.debian.org/security/2018/dsa-4307"
    ],
    "refmap": {
        "bid": [
            "105396"
        ],
        "confirm": [
            "https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-14647"
        ],
        "debian": [
            "DSA-4306",
            "DSA-4307"
        ],
        "misc": [
            "https://bugs.python.org/issue34623"
        ],
        "sectrack": [
            "1041740"
        ],
        "ubuntu": [
            "USN-3817-1",
            "USN-3817-2"
        ]
    },
    "summary": "Python's elementtree C accelerator failed to initialise Expat's hash salt during initialization. This could make it easy to conduct denial of service attacks against Expat by constructing an XML document that would cause pathological hash collisions in Expat's internal data structures, consuming large amounts CPU and RAM. Python 3.8, 3.7, 3.6, 3.5, 3.4, 2.7 are believed to be vulnerable.",
    "vulnerable_configuration": [
        {
            "id": "cpe:2.3:a:python:python:2.7.0",
            "title": "Python 2.7.0"
        },
        {
            "id": "cpe:2.3:a:python:python:2.7.1",
            "title": "Python 2.7.1"
        },
        {
            "id": "cpe:2.3:a:python:python:2.7.1:rc1",
            "title": "Python 2.7.1 Release Candiate 1"
        },
        {
            "id": "cpe:2.3:a:python:python:2.7.2",
            "title": "Python 2.7.2"
        },
        {
            "id": "cpe:2.3:a:python:python:2.7.2:rc1",
            "title": "Python 2.7.2 Release Candidate 1"
        },
        {
            "id": "cpe:2.3:a:python:python:2.7.3",
            "title": "Python 2.7.3"
        },
        {
            "id": "cpe:2.3:a:python:python:2.7.4",
            "title": "Python 2.7.4"
        },
        {
            "id": "cpe:2.3:a:python:python:2.7.5",
            "title": "Python 2.7.5"
        },
        {
            "id": "cpe:2.3:a:python:python:2.7.6",
            "title": "Python 2.7.6"
        },
        {
            "id": "cpe:2.3:a:python:python:2.7.7",
            "title": "Python 2.7.7"
        },
        {
            "id": "cpe:2.3:a:python:python:2.7.8",
            "title": "Python 2.7.8"
        },
        {
            "id": "cpe:2.3:a:python:python:2.7.9",
            "title": "Python 2.7.9"
        },
        {
            "id": "cpe:2.3:a:python:python:2.7.10",
            "title": "Python 2.7.10"
        },
        {
            "id": "cpe:2.3:a:python:python:2.7.11",
            "title": "Python 2.7.11"
        },
        {
            "id": "cpe:2.3:a:python:python:2.7.12",
            "title": "Python 2.7.12"
        },
        {
            "id": "cpe:2.3:a:python:python:2.7.13",
            "title": "Python 2.7.13"
        },
        {
            "id": "cpe:2.3:a:python:python:2.7.14",
            "title": "Python 2.7.14"
        },
        {
            "id": "cpe:2.3:a:python:python:2.7.15",
            "title": "Python 2.7.15"
        },
        {
            "id": "cpe:2.3:a:python:python:3.4.0",
            "title": "Python 3.4.0"
        },
        {
            "id": "cpe:2.3:a:python:python:3.4.1",
            "title": "Python 3.4.1"
        },
        {
            "id": "cpe:2.3:a:python:python:3.4.2",
            "title": "Python 3.4.2"
        },
        {
            "id": "cpe:2.3:a:python:python:3.4.3",
            "title": "Python 3.4.3"
        },
        {
            "id": "cpe:2.3:a:python:python:3.4.4",
            "title": "Python 3.4.4"
        },
        {
            "id": "cpe:2.3:a:python:python:3.4.5",
            "title": "Python 3.4.5"
        },
        {
            "id": "cpe:2.3:a:python:python:3.4.6",
            "title": "Python 3.4.6"
        },
        {
            "id": "cpe:2.3:a:python:python:3.4.7",
            "title": "Python 3.4.7"
        },
        {
            "id": "cpe:2.3:a:python:python:3.5.0",
            "title": "Python 3.5.0"
        },
        {
            "id": "cpe:2.3:a:python:python:3.5.1",
            "title": "Python 3.5.1"
        },
        {
            "id": "cpe:2.3:a:python:python:3.5.2",
            "title": "Python 3.5.2"
        },
        {
            "id": "cpe:2.3:a:python:python:3.5.3",
            "title": "Python 3.5.3"
        },
        {
            "id": "cpe:2.3:a:python:python:3.5.4",
            "title": "Python 3.5.4"
        },
        {
            "id": "cpe:2.3:a:python:python:3.5.5",
            "title": "Python 3.5.5"
        },
        {
            "id": "cpe:2.3:a:python:python:3.5.6",
            "title": "Python 3.5.6"
        },
        {
            "id": "cpe:2.3:a:python:python:3.6.0",
            "title": "Python 3.6.0"
        },
        {
            "id": "cpe:2.3:a:python:python:3.6.1",
            "title": "Python 3.6.1"
        },
        {
            "id": "cpe:2.3:a:python:python:3.6.2",
            "title": "Python 3.6.2"
        },
        {
            "id": "cpe:2.3:a:python:python:3.6.3",
            "title": "Python 3.6.3"
        },
        {
            "id": "cpe:2.3:a:python:python:3.6.4",
            "title": "Python 3.6.4"
        },
        {
            "id": "cpe:2.3:a:python:python:3.6.5",
            "title": "Python 3.6.5"
        },
        {
            "id": "cpe:2.3:a:python:python:3.6.6",
            "title": "Python 3.6.6"
        },
        {
            "id": "cpe:2.3:a:python:python:3.7.0",
            "title": "Python 3.7.0"
        },
        {
            "id": "cpe:2.3:o:canonical:ubuntu_linux:12.04:-:-:-:esm",
            "title": "Canonical Ubuntu Linux 12.04 ESM (Extended Security Maintenance)"
        },
        {
            "id": "cpe:2.3:o:canonical:ubuntu_linux:14.04:-:-:-:lts",
            "title": "Canonical Ubuntu Linux  14.04 LTS (Long-Term Support)"
        },
        {
            "id": "cpe:2.3:o:canonical:ubuntu_linux:16.04:-:-:-:lts",
            "title": "Canonical Ubuntu Linux 16.04  LTS (Long-Term Support)"
        },
        {
            "id": "cpe:2.3:o:canonical:ubuntu_linux:18.04:-:-:-:lts",
            "title": "cpe:2.3:o:canonical:ubuntu_linux:18.04:-:-:-:lts"
        },
        {
            "id": "cpe:2.3:o:debian:debian_linux:9.0",
            "title": "Debian Linux 9.0"
        }
    ],
    "vulnerable_configuration_cpe_2_2": [
        "cpe:/a:python:python:2.7.0",
        "cpe:/a:python:python:2.7.1",
        "cpe:/a:python:python:2.7.1:rc1",
        "cpe:/a:python:python:2.7.2",
        "cpe:/a:python:python:2.7.2:rc1",
        "cpe:/a:python:python:2.7.3",
        "cpe:/a:python:python:2.7.4",
        "cpe:/a:python:python:2.7.5",
        "cpe:/a:python:python:2.7.6",
        "cpe:/a:python:python:2.7.7",
        "cpe:/a:python:python:2.7.8",
        "cpe:/a:python:python:2.7.9",
        "cpe:/a:python:python:2.7.10",
        "cpe:/a:python:python:2.7.11",
        "cpe:/a:python:python:2.7.12",
        "cpe:/a:python:python:2.7.13",
        "cpe:/a:python:python:2.7.14",
        "cpe:/a:python:python:2.7.15",
        "cpe:/a:python:python:3.4.0",
        "cpe:/a:python:python:3.4.1",
        "cpe:/a:python:python:3.4.2",
        "cpe:/a:python:python:3.4.3",
        "cpe:/a:python:python:3.4.4",
        "cpe:/a:python:python:3.4.5",
        "cpe:/a:python:python:3.4.6",
        "cpe:/a:python:python:3.4.7",
        "cpe:/a:python:python:3.5.0",
        "cpe:/a:python:python:3.5.1",
        "cpe:/a:python:python:3.5.2",
        "cpe:/a:python:python:3.5.3",
        "cpe:/a:python:python:3.5.4",
        "cpe:/a:python:python:3.5.5",
        "cpe:/a:python:python:3.5.6",
        "cpe:/a:python:python:3.6.0",
        "cpe:/a:python:python:3.6.1",
        "cpe:/a:python:python:3.6.2",
        "cpe:/a:python:python:3.6.3",
        "cpe:/a:python:python:3.6.4",
        "cpe:/a:python:python:3.6.5",
        "cpe:/a:python:python:3.6.6",
        "cpe:/a:python:python:3.7.0",
        "cpe:/o:canonical:ubuntu_linux:12.04::~~esm~~~",
        "cpe:/o:canonical:ubuntu_linux:14.04::~~lts~~~",
        "cpe:/o:canonical:ubuntu_linux:16.04::~~lts~~~",
        "cpe:/o:canonical:ubuntu_linux:18.04::~~lts~~~",
        "cpe:/o:debian:debian_linux:9.0"
    ]
}